### PR TITLE
TLSEXT_SIGALG_gostr34102012 0x0840 and 0x0841 support

### PR DESCRIPTION
**Pull request by NMorozxov**
For GOST2012-GOST8912-GOST8912 IANA introduce signature Signature Algorithm parametrs
https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-16

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: ssl/ssl_local.h [hunk 1]
```diff
@@ -2115,6 +2115,8 @@ typedef enum downgrade_en {
 #define TLSEXT_SIGALG_rsa_pss_rsae_sha384                       0x0805
 #define TLSEXT_SIGALG_rsa_pss_rsae_sha512                       0x0806
 #define TLSEXT_SIGALG_rsa_pss_pss_sha256                        0x0809
+#define TLSEXT_SIGALG_gostr34102012_256_intrinsic               0x0840
```
Comment (t8m) [inline ssl/ssl_local.h:2118]: This is really a nit however why are you adding the gostr sigalgs in between rsa_pss ones?

File: ssl/t1_trce.c [hunk 2]
```diff
@@ -557,6 +557,8 @@ static const ssl_trace_tbl ssl_sigalg_tbl[] = {
     {TLSEXT_SIGALG_rsa_pss_rsae_sha384, "rsa_pss_rsae_sha384"},
     {TLSEXT_SIGALG_rsa_pss_rsae_sha512, "rsa_pss_rsae_sha512"},
     {TLSEXT_SIGALG_rsa_pss_pss_sha256, "rsa_pss_pss_sha256"},
+    {TLSEXT_SIGALG_gostr34102012_256_intrinsic, "gost2012_256"},
```
Comment (t8m) [inline ssl/t1_trce.c:560]: The same nit as above. Please put them along the other gost sigalgs.

File: ssl/ssl_local.h (hunk 1 above)
Comment (NMorozxov) [reply ssl/ssl_local.h:2118]: sorted by value

File: ssl/t1_trce.c (hunk 2 above)
Comment (NMorozxov) [reply ssl/t1_trce.c:560]: Ok, no problem i will but them along the gost in both places

File: ssl/ssl_local.h (hunk 1 above)
Comment (NMorozxov) [reply ssl/ssl_local.h:2118]: ohhh...
this list isn't sorted, i miss

**Comment by NMorozxov** [comment]
@t8m @beldmit fix commited

**Comment by beldmit** [comment]
> Dmitry, please re-approve.

Done.

**Comment by NMorozxov** [comment]
ping @beldmit @t8m

**Comment by beldmit** [comment]
I don't understand why the label was not applied automatically.

**Comment by mattcaswell** [comment]
As per the comment  from openssl-machine above "this PR has been updated in that time the label 'approval: ready to merge' is not being automatically set." - you added a comment after "approval done".

**Comment by beldmit** [comment]
This behavior seems a bit strange - no code changes/change requests... Thanks for the clarification!

**Comment by beldmit** [comment]
Merged. Thanks!

**Comment by mattcaswell** [comment]
> This behavior seems a bit strange - no code changes/change requests... Thanks for the clarification!

The bot doesn't know the difference between an innocuous "Looks good to me" comment from an approval status relevant "This code is rubbish, I think this PR should be closed without merge". So it errs on the side of caution. Any update whatsoever results in the tag not being applied automatically - instead you just get the message above.

**Pull request closed without merging**